optuna/storages/_rdb/storage.py

```diff
@@ -164,12 +167,6 @@ def objective(trial):
         You can use :class:`~optuna.storages.RetryFailedTrialCallback` to automatically retry
         failed trials detected by heartbeat.
 
-    Raises:
-        :exc:`ValueError`:
-            If the given `heartbeat_interval` or `grace_period` is not a positive integer.
-        :exc:`RuntimeError`:
-            When a process tries to finish a trial that has already
-            been set to :class:`~optuna.trial.TrialState.FAIL` by heartbeat.
```

Comment:
Can we move this explanation to some places? It is useful for users to note that the already failed trial by heartbeat cannot be finished by the original process.

Reply:
I'm not sure where to move this explanation. I'd like to ask @toshihikoyanase or @belldandyxtq when/where the error is raised. :bow:

https://github.com/optuna/optuna/pull/2802

Reply:
We discussed this privately and decided to simply remove it here.